conans/test/unittests/model/test_conf.py

```diff
@@ -85,3 +90,103 @@ def test_parse_spaces():
     c = ConfDefinition()
     c.loads(text)
     assert c["core:verbosity"] == "minimal"
+
+
+@pytest.mark.parametrize("text, expected", [
+    ("user.company.cpu:jobs=!", None),
+    ("user.company.cpu:jobs=10", 10),
+    ("user.company.build:ccflags=--m superflag", "--m superflag"),
+    ("zlib:user.company.check:shared=True", True),
+    ("zlib:user.company.check:shared_str='True'", '"True"'),
+    ("user.company.list:objs=[1, 2, 3, 4, 'mystr', {'a': 1}]", [1, 2, 3, 4, 'mystr', {'a': 1}]),
+    ("user.company.network:proxies={'url': 'http://api.site.com/api', 'dataType': 'json', 'method': 'GET'}",
+     {'url': 'http://api.site.com/api', 'dataType': 'json', 'method': 'GET'})
+])
+def test_conf_get_different_type_input_objects(text, expected):
+    """
+    Testing any possible Python-evaluable-input-format introduced
+    by the user
+    """
+    c = ConfDefinition()
+    c.loads(text)
+    assert c.get(text.split("=")[0]) == expected
+
+
+def test_compose_conf_complex():
+    """
+    Testing the composition between several ConfDefiniton objects and with
+    different value types
+    """
+    text = textwrap.dedent("""\
+        user.company.cpu:jobs=10
+        user.company.build:ccflags=--m superflag
+        zlib:user.company.check:shared=True
+        zlib:user.company.check:shared_str="True"
+        user.company.list:objs=[1, 2, 3, 4, 'mystr', {'a': 1}]
+        user.company.network:proxies={'url': 'http://api.site.com/api', 'dataType': 'json', 'method': 'GET'}
+    """)
+    c = ConfDefinition()
+    c.loads(text)
+    text = textwrap.dedent("""\
+        user.company.cpu:jobs=5
+        user.company.build:ccflags=--m otherflag
+        zlib:user.company.check:shared=!
+        zlib:user.company.check:shared_str="False"
+        user.company.list:objs+=[5, 6]
+        user.company.list:objs=+0
```

Comment:
Could this be confusing? When is append and when it is an extend?

Reply:
What happens if you do += to a dictionary?

Reply:
Adding more tests 😉

Reply:
Anyway, let me explain it. `append` and `extend` has the same syntax as environment:

`+= -> append`
`=+ -> prepend`

For now, only lists are able to do it. If you try it with a dict-like object is going to fail.

Reply:
Yes, my point is, if you have a list `[1,2,3]` and you do `user.company.list:objs=+0` you get `[0,1,2,3]`. That is totally expected but, if you have `[0,1,2,3]` and you do `user.company.list:objs+=[5, 6]`, what you get?

a) `[0,1,2,3,4,5,6]`
b) `[0,1,2,3,4,[5,6]]`

What if I append a dictionary? `user.company.list:objs+={foo: 5}`

a) `[0,1,2,3,4, {foo: 5}]`

So, looks like if you are appending a list, you extend, but if you append another thing, you append.

> For now, only lists are able to do it. If you try it with a dict-like object is going to fail.

Good, I understand. I would like to see a test.